src/main/java/org/csanchez/jenkins/plugins/kubernetes/AllContainersRunningPodWatcher.java

```diff
@@ -72,8 +84,42 @@ boolean areAllContainersRunning(Pod pod) {
         }
         for (ContainerStatus containerStatus : containerStatuses) {
             if (containerStatus != null) {
-                if (containerStatus.getState().getWaiting() != null) {
+                ContainerStateWaiting waitingState = containerStatus.getState().getWaiting();
+                if (waitingState != null) {
+                    String waitingStateMsg = waitingState.getMessage();
+                    if (waitingStateMsg != null && waitingStateMsg.contains("Back-off pulling image")) {
+                        LOGGER.log(Level.INFO, "Unable to pull Docker image");
+
+                        Jenkins jenkins = Jenkins.getInstanceOrNull();
+                        if (jenkins != null) {
+                            Queue q = jenkins.getQueue();
+                            for (Queue.Item item : q.getItems()) {
+                                Label itemLabel = item.getAssignedLabel();
+                                if (itemLabel != null && isCorrespondingLabels(itemLabel.getDisplayName(), pod.getMetadata().getName(), LOGGER)) {
+                                    String itemTaskName = item.task.getFullDisplayName();
+                                    String jobName = getJobName(itemTaskName);
+                                    if (jobName.equals("")) {
+                                        LOGGER.log(Level.WARNING, "Unknown / Invalid job format name");
+                                        break;
+                                    }
+                                    Run run = getCorrespondingJobBuild(jenkins.allItems(Job.class).iterator(), jobName, getBuildNumber(itemTaskName));
```

Comment:
Way too much work. Should suffice for `KubernetesLauncher.launch` to pass its `TaskListener` to the constructor of `AllContainersRunningPodWatcher`. Then you have a simple

```java
listener.error("Unable to pull Docker image. Check if image name is spelled correctly");
```

and you are done.

Reply:
The reason we needed to grab the corresponding Queue.Item was because if you simply error, the build will continue `Waiting for next available executor` instead of actually stopping. Because we're checking for a bad docker image, that means the user will have to go and change it in their pod template and start a new build anyways. So we wanted to cancel/stop the job for them instead of having it wait indefinitely.

Reply:
Passing the `KubernetesLauncher`'s `TaskListener` to `AllContainersRunningPodWatcher` doesn't work.  The error message doesn't get printed to the build console.

Reply:
Sorry, I missed the `Queue.cancel` call here. I see the idea but that is not really the right API to use here. In addition to printing a message to a `TaskListener`, just have `AllContainersRunningPodWatcher.await` throw a suitable exception, so that `KubernetesSlave.terminate` will be called by `KubernetesLauncher.launch`.